save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -126,11 +132,22 @@ class WarnPlugin(
             .groupBy { if (it.line != null && it.column != null) it.line to it.column else null }
             .mapValues { (_, warning) -> warning.sortedBy { it.message } }
 
-        return TestResult(
-            paths,
-            checkResults(expectedWarnings, actualWarningsMap, warnPluginConfig),
-            DebugInfo(stdout, stderr, null)
-        )
+        val testResultList: MutableList<TestResult> = mutableListOf()
+        paths.forEach { path ->
```

Comment:
why do you need forEach here with mutable list, use map instead in single functional manner